server/fleet/datastore.go

```diff
@@ -1261,10 +1264,22 @@ type Datastore interface {
 	// GetHostLockWipeStatus gets the lock/unlock and wipe status for the host.
 	GetHostLockWipeStatus(ctx context.Context, hostID uint, fleetPlatform string) (*HostLockWipeStatus, error)
 
+	// LockHostViaScript sends a script to lock a host and updates the
+	// states in host_mdm_actions
 	LockHostViaScript(ctx context.Context, request *HostScriptRequestPayload) error
+
+	// UnlockHostViaScript sends a script to unlock a host and updates the
+	// states in host_mdm_actions
 	UnlockHostViaScript(ctx context.Context, request *HostScriptRequestPayload) error
 }
 
+// MDMAppleStore wraps nanomdm's storage and adds methods to deal with
+// Fleet-specific use cases.
+type MDMAppleStore interface {
+	storage.AllStorage
+	EnqueueDeviceLockCommand(ctx context.Context, host *Host, cmd *mdm.Command, pin string) error
+}
```

Comment:
the idea of this wrapper is to give us the flexibility we'll need very soon to track things like: methods enqueued by Fleet vs an user, etc.